.github/workflows/build.yml

```diff
@@ -308,3 +310,194 @@ jobs:
           commit-message: publish documentation
           clean-exclude: |
             "version/*"
+
+  upload-nightly:
+    if: (github.ref == 'refs/heads/master') && (github.event_name == 'schedule')
+    needs: [test-unix, test-macos, test-windows]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/download-artifact@v2
+      with:
+        name: wheels
+        path: dist/
+
+    - run: |
+        zip -r dist.zip dist/
+
+    - uses: actions/github-script@v5
+      id: asset
+      with:
+        github-token: ${{ secrets.gh_access_token }}
+        script: |
+          const fs = require('fs');
+
+          // Get the ref for HEAD [A call to the API is probably useless]
+          let ref_head = await github.rest.git.getRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: 'heads/master',
+          });
+          console.log(`HEAD reference ${ref_head.data.object.sha}`);
+
+          // Retrieve ref for tag `nightly`
+          let ref_nightly = null;
+          try {
+            ref_nightly = await github.rest.git.getRef({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              ref: 'tags/nightly',
+            });
+
+            if (ref_nightly.data.object.sha === ref_head.data.object.sha) {
+              console.log('No new nightly release');
+              return;
```

Comment:
No need for console.log, you can replace by
```js
    return "No new nightly release";
```

Reply:
done